### PR TITLE
Load variables from bash script generated by cloud-init

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request:
* Combines the master and replica FreeIPA setup scripts into a single script.
* Reads non-sensitive variable values from a small bash file that will be generated by cloud-init.

## 💭 Motivation and Context

* I merged the master and replica setup scripts into a single file because doing so gets rid of some code duplication.
* I read the non-sensitive variable values from a bash file generated by cloud-init to make installation as easy and hands-off as possible.  The user still has to enter sensitive data such as the Kerberos password for the `admin` user.

See also cisagov/freeipa-server-tf-module#26 and cisagov/cool-sharedservices-freeipa#14.

## 🧪 Testing

I have used this code to successfully redeploy the FreeIPA server cluster in our staging environment.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: files/setup_freeipa.sh [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                               --realm="$realm" \
+                               --domain="$domain" \
+                               --hostname="$hostname" \
+                               --ip-address="$ip_address" \
+                               --no-ntp \
+                               --no_hbac_allow \
+                               --mkhomedir
+
+            # Get kerberos credentials and create the dhs_certmapdata
+            # rule
```
Comment (dav3r) [inline files/setup_freeipa.sh]: Can this comment be expanded slightly to explain why we add the `dhs_certmapdata` rule?

File: files/setup_freeipa.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    # Add a principal alias for the instance ID so folks can ssh in
+    # via SSM Session Manager.
+    ipa host-add-principal \
+        "$hostname" \
+        host/"$(curl http://169.254.169.254/latest/meta-data/instance-id)"."$domain"
+}
+
+
+if [ $# -lt 1 ]
```
Comment (dav3r) [inline files/setup_freeipa.sh]: Wouldn't `if [ $# -ne 1 ]` be better here?

**Review by dav3r — APPROVED** [review]
Very good old chap! 🎩 
I have but two brief comments for your consideration.

File: files/setup_freeipa.sh (hunk 1 above)
Comment (jsf9k) [reply files/setup_freeipa.sh]: See commit 660bf23.

File: files/setup_freeipa.sh (hunk 2 above)
Comment (jsf9k) [reply files/setup_freeipa.sh]: See commit 372efa8.

File: files/setup_freeipa.sh (hunk 1 above)
Comment (dav3r) [reply files/setup_freeipa.sh]: Thanks for adding that excellent explanation!

File: files/setup_freeipa.sh [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    ip route | grep default | sed "s/^.* dev \([^ ]*\).*$/\1/"
+}
+
+# Get the IP address corresponding to an interface
+function get_ip {
+    ip --family inet address show dev "$1" | \
+        grep inet | \
+        sed "s/^ *//" | \
+        cut --delimiter=' ' --fields=2 | \
+        cut --delimiter='/' --fields=1
```
Comment (felddy) [inline files/setup_freeipa.sh]: I feel like this might come back to visit us later.

Comment (felddy) [reply files/setup_freeipa.sh]: `ip --family inet address show dev "$1" | grep -Po 'inet \K[\d.]+'`

Explained: https://regexr.com/56khv

Stolen from: https://unix.stackexchange.com/questions/87468/is-there-an-easy-way-to-programmatically-extract-ip-address

**Review by felddy — APPROVED** [review]
Looks good!  See idea about IP extraction from `ip`.  🌶️

File: files/setup_freeipa.sh (hunk 3 above)
Comment (jsf9k) [reply files/setup_freeipa.sh]: See commit cefc6f7.

**Pull request merged**